router/core/context.go

```diff
@@ -272,6 +279,73 @@ type requestContext struct {
 	forceSha256Compute bool
 }
 
+type headerBuilder struct {
+	headers map[string]*HeaderWithHash
+	allHash uint64
+}
+
+func (c *headerBuilder) HashAll() (out uint64) {
+	return c.allHash
+}
+
+func (c *headerBuilder) HeadersForSubgraph(subgraphName string) (http.Header, uint64) {
+	if header, ok := c.headers[subgraphName]; ok {
+		return header.Header.Clone(), header.Hash
```

Comment:
One thing I found here, which might or might not be intended is:
It seems `header.Header` is nil for EDG's. I tested this and found this to be true at least for Kafka.
The reason seems to be that for EDG´s we dont have any headers (because it's not http connections to subgraphs but broker-specific connections we do right?). The function returns nil, 0 as the consequence. 

The `prepareTrigger` function in the engine does find a header builder and then adds `nil` to the hash input. It means the triggerID for EDFS subscriptions in the engine is based solely on the input, not headers. Again, could be by design. Just wanted to raise awereness in case this is unexpected behaviour.